### DISCUSSION TITLE
Do we get to know from IKE_UP or DOWN event it is due to reauth requested

### DISCUSSION THREAD

**Discussion by hbhokare** [strongswan/strongswan#528] [Q&A]
Hi,

We are trying to develop the IPSec based application using strongswan/davici library. We have following questions to handle certain situation in our application.

1. Do we get to know from IKE_UP or DOWN event,  event is caused due to reauth request indicating make before break policy or break before make policy. If yes then any specific field from event data indicates this ? 
2. if answer to above qustion is NO then how would we manage the ike sa objects at application layer when reauth is requested to strongswan/VICI ? Particulary in case of make before break policy because two valid ike sa exists point in time for an application.

Thanks in Advance !

**Comment by Thermi** [comment]
Please don't cross post.

**Reply by tobiasbrunner** [reply]
Where was this cross-posted?

**Reply by Thermi** [reply]
I assumed this was cross posted to the thread on the ML list regarding log and ike sa event order when initiating via vici.

**Comment by tobiasbrunner** [comment]
>     1. Do we get to know from IKE_UP or DOWN event,  event is caused due to reauth request indicating make before break policy or break before make policy.

No. But it's a global setting in strongswan.conf. On the other hand, you won't know at all that the events are generated because of a reauthentication (in particular on a responder that never knows this).

> if answer to above qustion is NO then how would we manage the ike sa objects at application layer when reauth is requested to strongswan/VICI ? Particulary in case of make before break policy because two valid ike sa exists point in time for an application.

Yes, you have to handle that situation. Or you don't use reauthentication but proper rekeying instead (of course, that only works if you have full control of the peers as there is no way to prevent them from using reauthentication if they want to).
